### PR TITLE
[instancer] support partial instancing for COLRv1

### PR DESCRIPTION
**Pull request by qxliu76**
This PR is dependent on #4679

### CODE REVIEW

**Comment by behdad** [comment]
Humm, I merged your other PR, now this has conflicts. Can you rebase please?

**Comment by qxliu76** [comment]
> Humm, I merged your other PR, now this has conflicts. Can you rebase please?

Rebased, thank you!

File: src/OT/Color/COLR/COLR.hh [hunk 1]
```diff
@@ -2219,6 +2308,88 @@ struct COLR
...
+    TRACE_SUBSET (this);
+    if (!varStore || c->plan->all_axes_pinned ||
+        !c->plan->colrv1_variation_idx_delta_map)
+      return_trace (true);
+
+    const ItemVariationStore& var_store = this+varStore;
+    if (c->plan->normalized_coords)
+    {
+      item_variations_t item_vars;
+      /* turn off varstore optimization when varIdxMap is null, so we maintain
```
Comment (behdad) [inline src/OT/Color/COLR/COLR.hh:2333]: This is probably fine. Ideally we should add a varIdxMap and optimize. Or measure the two and use whichever is more economical...

**Comment by behdad** [comment]
Is this ready?

**Comment by qxliu76** [comment]
> Is this ready?

Not yet, I'm checking on 1 test failure after rebase.

**Comment by qxliu76** [comment]
`distcheck` is clean, I think the PR is ready now.

**Pull request merged**